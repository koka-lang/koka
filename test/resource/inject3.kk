// Test inject over exceptions (should insert "inject-exn")
fun foo( action : () -> e int ) : e int {
  try {
    inject<exn>(action)  
    error("test")
    60
  }
  fun(exn) {
    52
  }
}

fun main() {
  try-default(42){ foo{ error("should throw") }}.println
}