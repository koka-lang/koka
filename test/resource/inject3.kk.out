42
foo : forall<e> (action : () -> e int) -> e int
main : () -> console ()